Separate class definitions in sdlmain.cpp to sdlmain.h

Wengier opened the pull request:
This moves some class definitions in sdlmain.cpp into a separate header file sdlmain.h, so that the header file can be included in other source files such as render.cpp for features like TTF support.

kcgen reviewed (changes requested):
Looks good @Wengier ; only a couple minor suggestions:

- all sources should have the usual GPL copyright block, so suggest adding that to the new header.  Recommend using the SPDX style, from one of the Staging files like `bitops.h`
- Suggest moving the PPScale class into the existing src/libs/ppscale/ppscale.h, and the class's constructor implementation into src/libs/ppscale/ppscale.cpp.

Wengier commented:
@kcgen Thanks for the suggestions! Added Staging header (just like ``bitops.h``) and moved PPScale class into ``src/libs/ppscale/ppscale.h`` accordingly. But I could not move the PPScale class's constructor implementation into ``src/libs/ppscale/ppscale.c`` (not ``ppscale.c``) because it is not a C++ file so the code could not compile in that file. So it is also in the ppscale.h file now.

kcgen commented:
Oh, good point regarding the C vs CPP file issue. 
Thanks @Wengier ; merging.
